### ISSUE TITLE
thanks for making this

### ISSUE THREAD

**Issue by vegarringdal** [ozum/knex-migrate-sql-file#1]
thanks for making this

I made some edits for my self
```js
const sqlFile = { up: getSQLFileName("01_up"), down: getSQLFileName("02_down") };
```
this way it was easier to read

![image](https://user-images.githubusercontent.com/2901416/41557497-2c318cfc-733e-11e8-9674-3d4155b120e0.png)

Also made a helper file so I could generate files
`node node generate-migration-file add_milestone_table`
generates 

![image](https://user-images.githubusercontent.com/2901416/41557565-5b0d51d2-733e-11e8-9cf6-9218d01e07c1.png)

sources for my generate file:
(no tests, just needed something to stop doing copy and paste errros)
```js
const path = require("path");
const fs = require("fs");



const writeFile = (file, data) => {
    return new Promise((resolve, reject) => {
        fs.appendFile(path.resolve('migrations', file), data, function (err) {
            if (err) {
                reject(err);
            } else {
                resolve(data);
            }
        });
    });
};

const MIGRATION_NAME = process.argv[2];
let DATE = new Date();
let YEAR = DATE.getFullYear();
let MONTH = DATE.getMonth() + 1;
let DAY = DATE.getDate() + 1;
let HOUR = DATE.getHours() + 1;
let MINUTE = DATE.getMinutes() + 1;
let SEC = DATE.getSeconds() + 1;

MONTH = MONTH > 9 ? MONTH : '0' + MONTH
DAY = DAY > 9 ? DAY : '0' + DAY
HOUR = HOUR > 9 ? HOUR : '0' + HOUR
MINUTE = MINUTE > 9 ? MINUTE : '0' + MINUTE
SEC = SEC > 9 ? SEC : '0' + SEC
const FILENAME = `${YEAR}${MONTH}${DAY}${HOUR}${MINUTE}${SEC}_${MIGRATION_NAME}`;


writeFile(`${FILENAME}.js`, `module.exports = require("../knex-migrate-sql-file.js")();`);
writeFile(`${FILENAME}.01_up.sql`, `--${FILENAME}-up`);
writeFile(`${FILENAME}.02_down.sql`, `--${FILENAME}-down`);
```

dunno if you need similar or have made something already, but figured I could share :-)

**Comment by ozum** [comment]
Thanks for sharing

**Comment by vegarringdal** [comment]
Looks like you have other postgres stuff too, nice.
Want to try and make simple app using postgres with node/kosjs on backend and hobby framework project on client side.
Very slow progres atm since I end up doing everything else all the time, and I need to learn more sql/postgres 😂 
https://github.com/vegarringdal/projectControl

Do you use vscode for with postgres?
If so have you seen a linter for sql ?

**Comment by ozum** [comment]
Yes I have other postgres modules. (See [pg-structure.com](http://www.pg-structure.com) which provides pg database structure in JS objects and [pg-generator.com](http://www.pg-generator.com) which generates files based on nunjucks templates)

I use vscode for js and ts, but not for postgres (besides simple sql scripts)

For pg I suggest [EMS SQL Manager](https://www.sqlmanager.net/en/). It has free (limited) and paid versions. It is for Windows, but I use it on Mac with Wine, because I can't find a replacement for it (I tried tens of applications)

If I don't use EMS, I may use those Mac native applications: (I don't know whether they have free versions): DataGrip or Navicat Premium.

**Comment by vegarringdal** [comment]
@ozum thx

I tried Datagrip but found pgAdmin4 better, found it very weird to use. 
Maybe I just did not try it enough.
Navicat looks very good. 

How is SQL Manager  vs Navicat ?
Whats the main reasons why you prefer one over the other ?

**Comment by ozum** [comment]
Paid version of EMS SQL Manager supports every single PostgreSQL feature (for example postgresql aggregates, user defined types etc.)

Navicat has some extra utilities (automation and some cloud services etc.) compared to EMS, and supports most of the postgresql features but not **every single** feature, which is no-go for me.

I found Datagrip also weird and hard to use, but considering it's a JetBrains product, it's code editor is far superior than EMS and Navicat (fast, better code completion etc.). Since you asked a linter I assumed it may be a better fit for you than it's for me.

I found pgAdmin's web like interface a little irritating.
